### DISCUSSION TITLE
Prevent custom entry element source from appearing in multi-site

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [craftcms/cms#12975] [Q&A] [answered]
Is there anyway to prevent an entry element source being registered by a module from appearing in the entries column of multi site? 

It seems using `Craft::$app->getSites()` won't work because it returns the default site.

The element source being registered by a module is for the default site, but I don't want it appearing on the other site. Is there any condition or method to prevent this?

**Comment by janhenckens** [comment]
Not sure how error-proof this is but in the last versions for Craft there is a URL parameter with the site from which entries are being displayed, maybe you can based it on that?

**Reply by jamesmacwhite** [reply]
Interesting idea, the problem is, I don't even think URL will help. The element sources change when using the multi-site dropdown within entries when switching between one or more site, this looks like a JS/AJAX related request, because it's not a request that reloads the entries tab, so I'm not sure how I can prevent the entry sources being included.

There must be some logic to control it, because sources like channels and other sources are removed when switching, but the module that's using the EVENT_REGISTER_SOURCES event has them appearing across any site.

**Comment by jamesmacwhite** [comment]
The only hack I've thought about is inserting a custom body class that ties to the multi-site being shown in the entries with some JS to update body each time and have CSS rules that hide the entry sources when not required, but I have feeling there must be a better way.

I can see a request to `manage/actions/element-indexes/get-elements` occurs when loading each multi-site on the entries screen, I have to look at this controller for more ideas.

**Comment by jamesmacwhite** [comment, marked as answer]
I'm an idiot, there's actually a super simple way to solve this.

On the event you can set a `sites` key, which can be set to the siteId required to limit it appearing. It can be set to a specific ID or an array of siteIds for multiple.

```php
$event->sources[] = [
	'key' => 'keyName',
	'sites' => 1,
	'label' => 'Source label',
	'criteria' => [
            ....
	],
	'data' => [
		'handle' => 'exampleHandle',
	],
];
```

**Reply by brandonkelly** [reply]
You got it!
